src/Maker/Security/MakeFormLogin.php

```diff
@@ -111,38 +116,44 @@ public function interact(InputInterface $input, ConsoleStyle $io, Command $comma
             throw new RuntimeCommandException(\sprintf('The file "%s" does not exist. PHP & XML configuration formats are currently not supported.', self::SECURITY_CONFIG_PATH));
         }
 
-        $this->ysm = new YamlSourceManipulator($this->fileManager->getFileContents(self::SECURITY_CONFIG_PATH));
-        $securityData = $this->ysm->getData();
+        $securityData = $this->getSecurityData();
 
         if (!isset($securityData['security']['providers']) || !$securityData['security']['providers']) {
             throw new RuntimeCommandException('To generate a form login authentication, you must configure at least one entry under "providers" in "security.yaml".');
         }
 
-        $this->controllerName = $io->ask(
-            'Choose a name for the controller class (e.g. <fg=yellow>SecurityController</>)',
-            'SecurityController',
-            Validator::validateClassName(...)
-        );
+        if (null === $input->getArgument('controllerName')) {
+            $input->setArgument(
+                'controllerName', $io->ask(
+                    'Choose a name for the controller class (e.g. <fg=yellow>SecurityController</>)',
+                    'SecurityController',
+                    Validator::validateClassName(...)
+                ));
+        }
 
-        $securityHelper = new InteractiveSecurityHelper();
-        $this->firewallToUpdate = $securityHelper->guessFirewallName($io, $securityData);
-        $this->userClass = $securityHelper->guessUserClass($io, $securityData['security']['providers']);
-        $this->userNameField = $securityHelper->guessUserNameField($io, $this->userClass, $securityData['security']['providers']);
-        $this->willLogout = $io->confirm('Do you want to generate a \'/logout\' URL?');
```

Comment:
Following on from the above, these all need to stay in interact. They should not be set in the `generate` method.